### ISSUE TITLE
Stream channel stops receiving new messages after app is backgrounded and then foregrounded

### ISSUE THREAD

**Issue by srgray** [GetStream/stream-chat-swift#1443]
## What did you do?
Backgrounded my chat app with a chat channel open. Waited ~30 seconds. Foreground app. Tried to receive new messages in chat channel

## What did you expect to happen?
I expected new messages in the chat channel to appear

## What happened instead?
New chat messages do not appear in the channel (but I can successfully send a message)

## GetStream Environment
**GetStream Chat version:** 4.0.0
**GetStream Chat frameworks:** StreamChat, StreamChatUI

## Additional context
See attached sample project:
[StreamBackgroundTest.zip](https://github.com/GetStream/stream-chat-swift/files/7150782/StreamBackgroundTest.zip)

Steps to reproduce:
- Run the sample app on 2 different devices/simulators
- Tap 'Join Chat as User 1' on 1 device and 'Join Chat as User 2' on the other
- Notice you send send/receive new chat messages from both users
- Background the app for User 1 for ~30 seconds
- Foreground the app for User 1
- Send a message from User 2: Notice User 1 does not see this new message

Do I need to do some reconnect logic to handle this case?

**Comment by tbarbugli** [comment]
@srgray can you update to latest v4 and retest?

**Comment by srgray** [comment]
@tbarbugli @evsaev This is still occurring with latest v4 SDK

**Comment by evsaev** [comment]
Hey @srgray, sorry for the delay and thanks for your patience! 🙌

The issue you're facing with is addressed by https://github.com/GetStream/stream-chat-swift/pull/1440. I'll let you know once it is merged.

**Comment by abdelrahman-badary** [comment]
@evsaev any updates on this, I noticed that the PR that had the proposed fix is closed?

**Comment by abdelrahman-badary** [comment]
@tbarbugli any updates about his?

**Comment by b-onc** [comment]
Hello @abdelrahman-badary (and also @srgray )

This issue happens because SDK does not call `synchronize` implicitly on reconnection. This, in turn, causes WS events to not arrive. This issue doesn't happen when a `ChannelListController` is used, since that component will implicitly call `watch` on channels.

The workaround is to subclass `ChannelVC` and call `synchronize` explicitly:
```swift
class CustomChannelVC: ChatChannelVC {
    override func viewDidLoad() {
        super.viewDidLoad()
        
        NotificationCenter.default.addObserver(
            self,
            selector: #selector(handleAppDidBecomeActive),
            name: UIApplication.didBecomeActiveNotification,
            object: nil
        )
    }
    
    @objc func handleAppDidBecomeActive() {
        channelController.synchronize()
    }
}
```

This will unblock you for the time being. We're working on a fix and we'll get it our as soon as possible.

**Comment by polqf** [comment]
Hello! We just released 4.12.0, you can check the changelog [here](https://github.com/GetStream/stream-chat-swift/releases/tag/4.12.0).
This issue should be fixed now. Feel free to reopen it if needed
